JPetCommonTools/JPetCommonTools.cpp

```diff
@@ -65,3 +65,52 @@ std::vector<const char*> JPetCommonTools::createArgs(const std::string& commandL
   });
   return args_char;
 }
+
+ std::string JPetCommonTools::extractDataTypeFromFile(const std::string& filename)
+{
+  auto file = extractFileNameFromFullPath(filename);
+
+  auto pos = file.find(".");
```

Comment:
In general I think one could use boost filename to perform all those operations, but it is ok for me.
Maybe I would define at the beginning "root" string cuse you use it.

Reply:
I believe our way of indicating the data type by a set of dot-separated words between the filename and the actual extension is quite non-standard (as this is neither filename nor extension in the standard understanding) and may not be handled by boost::filesystem. 
Where are the methods we already have?
I do not understand the point about requiring that the file exists... do these new methods require that?

Reply:
Yes, I agree it is not standard, Leave it as it is, it is minor thing.
the last comment is just to say that boost filename methods does not rely on the existence of the file in case you would like to use it.